zerver/models.py

```diff
@@ -918,6 +918,13 @@ class UserProfile(AbstractBaseUser, PermissionsMixin):
     # https://docs.djangoproject.com/en/1.10/ref/models/fields/#django.db.models.Field.null.
     timezone = models.CharField(max_length=40, default=u'')  # type: str
 
+    # user list sidebar
+    ALL_USERS           = 'all'
+    STREAM_MEMBERS      = 'member'
+    USER_LIST_CONTENT_CHOICES   = ((ALL_USERS, "All users"),
+                                   (STREAM_MEMBERS, "Stream or PM recipients"))
+    user_list_content = models.CharField(default=ALL_USERS, choices=USER_LIST_CONTENT_CHOICES, max_length=10)  # type: str
```

Comment:
We should do this as an enum-type field, rather than a string-type field.  `demote_inactive_streams` is a good example of the right structure.

(There's a lot of disadvantages to strings, varying from consuming a lot more storage in the database, to limits on the length of names to future options, to more risk of typos creating weird behavior)

Reply:
Maybe this should be called `buddy_list_style`?  And then the values would be
```
BUDDY_LIST_STYLE_ALL_USERS = 1
BUDDY_LIST_STYLE_MEMBERS = 2
```

Reply:
maybe BUDDY_LIST_STYLE_RECIPIENTS? 
When I see "all users vs members" I jump to "all users includes guests, and members doesn't".

Reply:
is `style` really the best we could do here?
I used `content` above but why not use something like `mode`. (although that only semantically makes sense as a suffix...)

so:
```
BUDDY_LIST_ALL_USERS_MODE = 1
BUDDY_LIST_MEMBERS_MODE = 2
```

or maybe the prefix `RECIPIENT_MODE`?

I'll wait on changing the modal until we figure out the name :sweat_smile:.